### PR TITLE
feat!: new components api

### PR DESCRIPTION
**Pull request by cbt1**
Updates the API to adhere to the new components api defined in #404

### CODE REVIEW

File: api-specifications/properties.json [hunk 1]
```diff
@@ -229,70 +229,229 @@
...
-        "hoverFontColor": {
-          "description": "When only `hoverColor` is set, this is adjusted to either `#f4f4f4` or `#ffffff` for optimal contrast",
+        "background": {
+          "description": "Cell background color palette",
           "optional": true,
           "type": "#/definitions/PaletteColor"
+        }
+      }
+    },
+    "PartialCellStyling": {
```
Comment (cbt1) [inline api-specifications/properties.json:411]: No idea what to call this. Maybe there is a better way to pick some props from another type?

File: src/qae/initial-properties.js [hunk 2]
```diff
@@ -117,34 +117,98 @@ const properties = {
...
+ */
+
+/**
+ * Mandatory key for chart styling
+ * @name ChartStylingKey
+ * @type {"theme"}
+ */
+
+/**
+ * Custom styling of cells
```
Comment (cbt1) [inline src/qae/initial-properties.js:171]: Maybe change to just `Chart styling`?

File: src/qae/initial-properties.js [hunk 3]
```diff
@@ -117,34 +117,98 @@ const properties = {
...
+ * @type object
+ * @property {GeneralStylingKey} key
+ * @property {TitleStyling} title
+ */
+
+/**
+ * Title styling options
+ * @name TitleOptions
+ * @type object
+ * @property {string} [fontSize] - Font size in pixel value
```
Comment (haxxmaxx) [inline src/qae/initial-properties.js:149]: I don't recognize this syntax, are the [] to highlight the prop name?

File: api-specifications/properties.json (hunk 1 above)
Comment (haxxmaxx) [reply api-specifications/properties.json:411]: Yeah you could reuse CellStyling and omit the props not needed. But you still would have to name it right? I think the current name is fine.

**Review by haxxmaxx — APPROVED** [review]

File: api-specifications/properties.json (hunk 1 above)
Comment (cbt1) [reply api-specifications/properties.json:411]: Yes I don't think there is a way around that

File: src/qae/initial-properties.js (hunk 3 above)
Comment (cbt1) [reply src/qae/initial-properties.js:149]: `@property {string} [fontSize]` means that `fontSize` is optional. Which is the JSDoc way of doing it. You can also do @property {string=} fontSize which would be the Google Closure compiler syntax, that also works.

File: src/qae/initial-properties.js (hunk 2 above)
Comment (haxxmaxx) [reply src/qae/initial-properties.js:183]: yeah I think so, the grid stuff is not really applied to independent cells

File: src/qae/initial-properties.js (hunk 3 above)
Comment (haxxmaxx) [reply src/qae/initial-properties.js:161]: ah ok, didn't know that. I think I prefer this way

**Pull request merged**